types/src/transaction/authenticator.rs

```diff
@@ -354,7 +354,7 @@ impl AccountAuthenticator {
     }
 }
 
-/// A struct that represents an account authentication key. An account's address is the last 16
+/// A struct that represents an account authentication key. An account's address is the last `AccountAddress::LENGTH`
```

Comment:
@davidiw, an account's address is **set to be** the authentication key used to create it right? Because right now we assume `|auth_key| == |addr|`.

Reply:
muahaha

Reply:
sorry I got distracted...

* There's no derived address -- it is just the same as the authentication key
* There's no prefix -- it is always []

We should clean up the documentation and the code. The AuthenticationKey is now just a way to derive an account address from a source of authentication (or something). I'd rather do a proper clean up rather than a single nit.